pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -730,3 +740,17 @@ func checkSeedAssociations(ctx context.Context, c client.Client, seedName string
 
 	return nil
 }
+
+func mustEnableVPA(ctx context.Context, c client.Client, shoot *gardencorev1beta1.Shoot) (bool, error) {
+	if err := c.Get(ctx, kutil.Key(shoot.Status.TechnicalID, "vpa-admission-controller"), &appsv1.Deployment{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			// VPA deployment in shoot namespace was not found, so we have to enable the VPA for this seed until it's
+			// being deployed.
+			return true, nil
+		}
+		return false, err
+	}
+
+	// VPA deployment in shoot namespace was found, so we don't need to enable the VPA for this seed.
+	return false, nil
```

Comment:
Shouldn't we check for a `deletionTimestamp` in case a deployment was found? Maybe the deployment is not gone, yet.

Reply:
We can do that, but we don't do it on other places because the `Deployment` is not backed by a finalizer, i.e., it will be immediately removed from the system once you delete it (at least how we do it), even if there is still the replicaset/pods.

Reply:
Okay, valid point. Thanks!